plugins/module_utils/consul.py

```diff
@@ -68,6 +69,26 @@ def camel_case_key(key):
     return "".join(parts)
 
 
+def validate_check(check):
+    validate_duration_keys = ['Interval', 'Ttl', 'Timeout']
+    validate_tcp_regex = r"(?P<host>.*):(?P<port>(?:[0-9]+))$"
+    if 'Tcp' in check and check['Tcp'] is not None:
+        match = re.match(validate_tcp_regex, check['Tcp'])
+        if not match:
+            raise Exception('tcp check must be in host:port format')
+    for duration in validate_duration_keys:
+        if duration in check and check[duration] is not None:
+            check[duration] = validate_duration(check[duration])
+
+
+def validate_duration(duration):
+    if duration:
+        duration_units = ['ns', 'us', 'ms', 's', 'm', 'h']
+        if not any(duration.endswith(suffix) for suffix in duration_units):
```

Comment:
This is a bit weird to check. and would allow for durations like `5xms`. Maybe a regex check like `\d+(?:ns|us|ms|s|m|h)` would be better?